### PR TITLE
Update API Mocks with Universal Mocks

### PR DESCRIPTION
**Pull request by bbengfort**
Implements bufconn-style universal mocks for the remote peer (counterparty) and the GDS.

### CODE REVIEW

**Comment by pdeziel** [comment]
@bbengfort It looks like this is breaking the older peers mock (see lint errors), should we throw out the older mock or try to refactor the code to use the mock you just added? It looks like we will have to add a trisanetworkclient mock to do option 2.

**Comment by bbengfort** [comment]
> @bbengfort It looks like this is breaking the older peers mock (see lint errors), should we throw out the older mock or try to refactor the code to use the mock you just added? It looks like we will have to add a trisanetworkclient mock to do option 2.

I've thrown out the older mock and refactored the tests; thanks for reviewing it!

**Comment by bbengfort** [comment]
@pdeziel @DanielSollis -- I finalized the last test; this is ready for review and should also have all the code needed as an example for the FromContext tests.

**Review by pdeziel — APPROVED** [review]
Makes sense to me, thanks for putting in the legwork to adapt these tests so that we have a good template for the other repos.

File: pkg/trisa/peers/peer_test.go [hunk 1]
```diff
@@ -40,8 +91,9 @@ func TestExchangeKeys(t *testing.T) {
 	})
 
 	// Signing key should be overwritten
+	// TODO: please validate these checks are correctly checking what we expect
 	key = p.Info().SigningKey
-	require.NotEqual(t, &rsa.PublicKey{}, key)
+	require.NotEqual(t, signingKey.PublicKey, key)
```
[the remark this answers is not in the document]
Comment (bbengfort) [reply pkg/trisa/peers/peer_test.go:96]: That makes sense! Thanks for handling it in the follow up!

File: pkg/trisa/peers/peers_test.go [hunk 2]
```diff
@@ -4,58 +4,68 @@ import (
...
+	err = cache.Add(&peers.PeerInfo{})
+	require.EqualError(t, err, "common name is required for all peers")
 
+	// Generate a random key for some of our fixtures.
 	privateKey, err := rsa.GenerateKey(rand.Reader, 2048)
 	require.NoError(t, err)
 
-	// Test adding peers concurrently
+	// Test adding peers concurrently; the leonardo peer should be updated with
+	// consecutive updates to
```
[the remark this answers is not in the document]
Comment (bbengfort) [reply pkg/trisa/peers/peers_test.go:42]: Erm, yeh - whoops. Would you mind also handling this comment in the follow up? I could finish it, but I think the comment will change when you do your update.

**Pull request merged**